sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -431,18 +498,6 @@ public Size apply(MethodVisitor mv, Implementation.Context implementationContext
             new Compound(MethodInvocation.invoke(createUserCodeException), Throw.INSTANCE)
                 .apply(mv, implementationContext);
 
-        mv.visitLabel(catchBlockEnd);
-        // The frame contents after the try/catch block is the same
-        // as it was before.
-        mv.visitFrame(
```

Comment:
I think that removing this will create problems for the exception stack traces (could be wrong). I'm guessing this is what was giving you problems -- something about the frame (which is used for stack unwinding) disagreed with the code that was added in the block? Maybe there were some local variables/stack variables added within the block that needed to be unwound?